### PR TITLE
fix: ng-add @ama-sdk/schematics and override tree

### PR DESCRIPTION
**Pull request by cpaulve-1A**
## Related issues

- :bug: Fixes #(issue)
- :rocket: Feature #(issue)

### CODE REVIEW

**Review by matthieu-crouzet — APPROVED** [review]

**Review by pginoux-1A — DISMISSED** [review]

**Review by fpaul-1A — APPROVED** [review]

**Review by mrednic-1A — DISMISSED** [review]

**Review by vscaiceanu-1a — DISMISSED** [review]

**Review by vscaiceanu-1a — APPROVED** [review]

File: packages/@ama-sdk/schematics/schematics/typescript/core/schema.json [hunk 1]
```diff
@@ -12,6 +12,11 @@
         "$source": "argv",
         "index": 0
       }
+    },
+    "specConfigPath": {
+      "type": "string",
+      "description": "Spec config file\nMore info: https://openapi-generator.tech/docs/generators/typescript/#config-options",
+      "default": null
```
Comment (kpanot) [inline packages/@ama-sdk/schematics/schematics/typescript/core/schema.json:19]: same comment: should prefer `""` for Nx CLI

File: packages/@ama-sdk/schematics/schematics/ng-add/index.ts [hunk 2]
```diff
@@ -198,7 +200,7 @@ export function ngAdd(): Rule {
     updatePackageJsonScripts,
     (t) => {
       const packageJson = tree.readText(packageJsonPath);
-      const needsToInstallOpenApiGeneratorCli = packageJson.match(/@ama-sdk\/schematics:typescript-.*--spec-path ([.\\/a-zA-Z-]+) /)?.[1];
+      const needsToInstallOpenApiGeneratorCli = !!packageJson.match(/@ama-sdk\/schematics:typescript-/);
```
Comment (fpaul-1A) [inline packages/@ama-sdk/schematics/schematics/ng-add/index.ts:203]:
```suggestion
      const needsToInstallOpenApiGeneratorCli = /@ama-sdk\/schematics:typescript-/.test(packageJson);
```

File: packages/@ama-sdk/schematics/schematics/ng-add/index.ts [hunk 3]
```diff
@@ -164,17 +166,17 @@ const registerPackageSchematics = async (tree: Tree, context: SchematicContext)
     return () => tree;
   }
   const amaSdkSchematicsPackageJsonContent = JSON.parse(readFileSync(path.resolve(__dirname, '..', '..', 'package.json'), {encoding: 'utf-8'})) as PackageJson;
-  const amaSdkSchematicsVersion = amaSdkSchematicsPackageJsonContent.version ? `@${amaSdkSchematicsPackageJsonContent.version}` : '';
+  const amaSdkSchematicsVersion = amaSdkSchematicsPackageJsonContent.version?.replace('v', '');
```
Comment (fpaul-1A) [inline packages/@ama-sdk/schematics/schematics/ng-add/index.ts:169]:
```suggestion
  const amaSdkSchematicsVersion = amaSdkSchematicsPackageJsonContent.version?.replace(/^v/, '');
```

**Review by fpaul-1A — APPROVED** [review]

**Review by pginoux-1A — DISMISSED** [review]

File: packages/@ama-sdk/schematics/schematics/typescript/core/schema.ts [hunk 4]
```diff
@@ -3,4 +3,5 @@ import type { JsonObject } from '@angular-devkit/core';
 export interface NgGenerateTypescriptSDKCoreSchematicsSchema extends JsonObject {
   /** Path to the swagger specification used to generate the SDK */
   specPath: string;
+  specConfigPath: string | null;
```
Comment (kpanot) [inline packages/@ama-sdk/schematics/schematics/typescript/core/schema.ts:6]: no `| null`

**Review by kpanot — APPROVED** [review]

**Review by fpaul-1A — APPROVED** [review]

**Pull request merged**